doc/ui-guidelines.md

```diff
@@ -0,0 +1,254 @@
+# UI components coding guidelines
+
+> [!IMPORTANT]
+> React apps are made out of components. A component is a piece of the UI (user interface) that has its own logic and appearance. A component can be as small as a button, or as large as an entire screen.
+> React components are JavaScript functions that return markup
+> This document will provide best practices on how to write efficient React components in ClojureScript
+
+
+At the time of creating the Status app, the Reagent library was a solid choice. Back then, hooks didn't exist, and there weren't any libraries providing effective global state management. After Reagent's emergence, another library called Re-frame built upon Reagent. Together, they offered powerful tools for developing React applications with ClojureScript. However, as React evolved, significant changes occurred. Class components, utilized in Reagent, became deprecated. Instead, functional components and hooks emerged for state management. In Status 2.0, we began incorporating more functional components and hooks, resulting in a blend of both approaches. To simplify matters and reduce confusion, we opted to transition to functional components and hooks for local state management.
+
+BEFORE:
+```clojure
+(defn- view-internal
+  [_ _]
+  (let [pressed? (reagent/atom false)]
+    (fn
+      [{:keys [theme on-press on-long-press icon]}]
+      [rn/pressable
+       {:style               (style/main @pressed? theme)
+        :on-press            on-press
+        :on-press-in         #(reset! pressed? true)
+        :on-press-out        #(reset! pressed? nil)
+        :on-long-press       on-long-press}
+       [quo.icons/icon icon]])))
+
+(def view (theme/with-theme view-internal))
+```
+
+NOW:
+```clojure
+(defn view
+  [{:keys [on-press on-long-press icon]}]
+  (let [[pressed? set-pressed] (rn/use-state false)
+        theme                  (theme/use-theme-value)
+        on-press-in            (rn/use-callback #(set-pressed true))
+        on-press-out           (rn/use-callback #(set-pressed nil))]
+    [rn/pressable
+     {:style               (style/main pressed? theme)
+      :on-press            on-press
+      :on-press-in         on-press-in
+      :on-press-out        on-press-out
+      :on-long-press       on-long-press}
+     [quo.icons/icon icon]]))
+```
+
+
+- We no longer need to create an anonymous function for rendering. This removes unnecessary confusion and the need for specific knowledge on how it works and why it was needed.
+- `rn/use-state` is used instead of `reagent/atom`
+- State values no longer need to be dereferenced; they are accessible as regular symbols. This eliminates a common bug where the "@" symbol was inadvertently omitted.
+- `theme/with-theme` wrapper is not needed anymore, `(theme/use-theme-value)` hook can be used directly in the components
+- `:f>` not needed anymore, all components are functional by default
```

Comment:
@flexsurfer, could you please elaborate on this?
Reagent creates class components by default, and `:f>` is used to create functional ones. So as far as I understand, we still need to use `:f>` on `view` function from `NOW` example to make it functional component. Unless we [configure](https://github.com/reagent-project/reagent/blob/master/doc/ReagentCompiler.md) reagent hiccup-compiler which I think we didn't yet.
Or I missing smth?

Reply:
hey @vkjr yes you're right, this PR will be merged together with https://github.com/status-im/status-mobile/pull/18820

Reply:
ah, thanks)

Reply:
probably best to put this comment in the pr description as others might have missed that detail too 👍